pkg/morph/deploy/notary.go

```diff
@@ -871,3 +841,286 @@ func makeUnsignedDesignateCommitteeNotaryTx(roleContract *rolemgmt.Contract, com
+
+	err := committeeMultiSigAcc.ConvertMultisig(committeeMultiSigM, prm.committee)
+	if err != nil {
+		return fmt.Errorf("compose committee multi-signature account: %w", err)
+	}
+
+	committeeMultiSigAccID := committeeMultiSigAcc.ScriptHash()
+	chNotaryRequests := make(chan *result.NotaryRequestEvent, 100) // secure from blocking
+	// cache processed operations: when main transaction from received notary
+	// request is signed and sent by the local account, we receive the request from
+	// the channel again
+	mProcessedMainTxs := make(map[util.Uint256]struct{})
+
+	subID, err := prm.blockchain.ReceiveNotaryRequests(&neorpc.TxFilter{
+		Signer: &committeeMultiSigAccID,
+	}, chNotaryRequests)
+	if err != nil {
+		return fmt.Errorf("subscribe to notary requests from committee: %w", err)
+	}
+
+	go func() {
+		defer func() {
+			err := prm.blockchain.Unsubscribe(subID)
+			if err != nil {
+				prm.logger.Warn("failed to cancel subscription to notary requests", zap.Error(err))
+			}
+		}()
+
+		prm.logger.Info("listening to committee notary requests...")
+
+		for {
+			select {
+			case <-ctx.Done():
+				prm.logger.Info("stop listening to committee notary requests (context is done)", zap.Error(ctx.Err()))
+				return
+			case notaryEvent, ok := <-chNotaryRequests:
+				if !ok {
+					prm.logger.Info("stop listening to committee notary requests (subscription channel closed)")
+					return
+				}
+
+				// for simplicity, requests are handled one-by one. We could process them in parallel
+				// using worker pool, but actions seem to be relatively lightweight
+
+				const expectedSigners = 3 // sender + committee + Notary
+				mainTx := notaryEvent.NotaryRequest.MainTransaction
+				// note: instruction above can throw NPE and it's ok to panic: we confidently
+				// expect that only non-nil pointers will come from the channel (NeoGo
+				// guarantees)
+
+				srcMainTxHash := mainTx.Hash()
+				_, processed := mProcessedMainTxs[srcMainTxHash]
+
+				switch {
+				case processed:
+					prm.logger.Info("main transaction of the notary request has already been processed, skip",
+						zap.Stringer("ID", srcMainTxHash))
+					continue
+				case notaryEvent.Type != mempoolevent.TransactionAdded:
+					prm.logger.Info("unsupported type of the notary request event, skip",
+						zap.Stringer("got", notaryEvent.Type), zap.Stringer("expect", mempoolevent.TransactionAdded))
+					continue
+				case len(mainTx.Signers) != expectedSigners:
+					prm.logger.Info("unsupported number of signers of main transaction from the received notary request, skip",
+						zap.Int("expected", expectedSigners), zap.Int("got", len(mainTx.Signers)))
+					continue
+				case !mainTx.HasSigner(committeeMultiSigAccID):
+					// in theory, there can be any notary requests besides those sent by the current
+					// auto-deploy procedure. However, it's better to log with 'info' severity since
+					// this isn't really expected in practice.
```

Comment:
> this isn't really expected in practice

It's not true, change this comment. You can (and you likely will) start the auto-update procedure on the working network. Working network assumes that notary requests can appear from time to time. E.g. notary request on container put - this notary request is also signed by the committee and can also have three signers. Is it possible that some user will deside to cal `container put` during this integrated contract update procedure? Yes.